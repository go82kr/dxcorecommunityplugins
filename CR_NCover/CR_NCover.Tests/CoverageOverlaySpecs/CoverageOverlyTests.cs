using System.Collections.Generic;
using ApprovalTests;
using CheckPoint;
using DevExpress.CodeRush.Core;
using NUnit.Framework;
using TypeMock.ArrangeActAssert;
using Monitor=CheckPoint.Monitor;

namespace CR_NCover.Tests.CoverageOverlaySpecs
{
	[TestFixture]
	[Isolated]
	public class CoverageOverlyTests
	{
		[NUnit.Framework.SetUp]
		public void Setup()
		{
			visualStudioSentry = Monitor.Calls<VisualStudio>();

			overlay = new CoverageOverlay();
			Isolate.WhenCalled(() => CodeRush.TextViews.Active).ReturnRecursiveFake();
		}

		[NUnit.Framework.TearDown]
		public void TearDown()
		{
			Isolate.CleanUp();
		}

		private CoverageOverlay overlay;
		private Sentry visualStudioSentry;

		[NUnit.Framework.Test]
		public void DisplayOverlayOnShow()
		{
			overlay.Show(new List<CoverageResult> {new CoverageResult {VisitCount = 0}, new CoverageResult {VisitCount = 1}});

			Approvals.Approve(visualStudioSentry.Report);
		}

		[NUnit.Framework.Test]
		public void InvalidateOnHide()
		{
			overlay.Hide();

			Approvals.Approve(visualStudioSentry.Report);
		}
	}
}