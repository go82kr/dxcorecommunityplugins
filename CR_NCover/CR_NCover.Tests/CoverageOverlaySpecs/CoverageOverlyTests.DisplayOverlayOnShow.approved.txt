CR_NCover.VisualStudio.GetRectangleFromRange()
CR_NCover.VisualStudio.FillRectangle()
CR_NCover.VisualStudio.GetRectangleFromRange()
CR_NCover.VisualStudio.FillRectangle()
